packages/internal/bridge/sdk/src/lib/tenderly.test.ts

```diff
@@ -0,0 +1,291 @@
+    const simulations:TenderlySimulation[] = [
+      { from: '0x123', to: '0x456' },
+    ];
+
+    const axiosDataParameter = generateAxiosData(simulations, undefined);
+
+    const expectedResponse = {
+      id: 0,
+      jsonrpc: '2.0',
+      result: [
+        {
+          gas: '0x30ff2',
+          gasUsed: '0x2e496',
+        },
+        {
+          gas: '0x40ff2',
+          gasUsed: '0x2e496',
+        },
+      ],
+    };
+
+    mockedAxios.post.mockResolvedValueOnce({ data: expectedResponse });
+
+    await expect(
+      submitTenderlySimulations(chainId, simulations),
+    ).rejects.toThrowError(
+      'Estimating gas failed with mismatched responses',
+    );
+    expect(axios.post).toHaveBeenCalledTimes(1);
+    expect(axios.post).toHaveBeenCalledWith(
+      apiEndpoint,
+      axiosDataParameter,
+      axiosHeaders,
+    );
+  });
+
+  it('should throw an error if the response contains an error for a single simulation', async () => {
+    const chainId = '123'; // 123 will fallback to dev
+    const apiEndpoint = getTenderlyEndpoint(chainId);
+    const simulations:TenderlySimulation[] = [
+      { from: '0x123', to: '0x456' },
+    ];
+
+    const axiosDataParameter = generateAxiosData(simulations, undefined);
+    const errorMessage = 'this is an error!!';
+
+    const expectedResponse = {
+      id: 0,
+      jsonrpc: '2.0',
+      result: [
+        {
+          error: {
+            message: errorMessage,
+          },
+        },
+      ],
+    };
+
+    mockedAxios.post.mockResolvedValueOnce({ data: expectedResponse });
+
+    await expect(
+      submitTenderlySimulations(chainId, simulations),
+    ).rejects.toThrowError(
+      `Estimating deposit gas failed with the reason: ${errorMessage}`,
+    );
+    expect(axios.post).toHaveBeenCalledTimes(1);
+    expect(axios.post).toHaveBeenCalledWith(
+      apiEndpoint,
+      axiosDataParameter,
+      axiosHeaders,
+    );
+  });
+
+  it('should throw an error if the response contains an error for one of multiple simulations', async () => {
+    const chainId = '123'; // 123 will fallback to dev
+    const apiEndpoint = getTenderlyEndpoint(chainId);
+    const simulations:TenderlySimulation[] = [
+      { from: '0x123', to: '0x456' },
+      { from: '0x456', to: '0x789' },
+    ];
+
+    const axiosDataParameter = generateAxiosData(simulations, undefined);
+    const errorMessage = 'this is an error!!';
+
+    const expectedResponse = {
+      id: 0,
+      jsonrpc: '2.0',
+      result: [
+        {
+          gas: '0x40ff2',
+          gasUsed: '0x2e496',
+        },
+        {
+          error: {
+            message: errorMessage,
+          },
+        },
+      ],
+    };
+
+    mockedAxios.post.mockResolvedValueOnce({ data: expectedResponse });
+
+    await expect(
+      submitTenderlySimulations(chainId, simulations),
+    ).rejects.toThrowError(
+      `Estimating deposit gas failed with the reason: ${errorMessage}`,
```

Comment:
nit: should we remove reference to "deposit" in this error message, given this could relate to a withdrawal transaction where we are estimating L1 gas?